README.md

```diff
@@ -118,3 +113,221 @@ This stack is IPv6-capable. Caveats:
         -A FORWARD -d 2a01:4f8:a0:12eb:deec::/80 -i eth0 -j REJECT --reject-with icmp6-port-unreachable
+
+Development: Getting Started Guide
+----------------------------------
+
+As desec-stack utilizes a number of different technologies and software packages, it requires some effort to setup a stack ready for development.
+While there are certainly many ways to get started hacking desec-stack, here is one way to do it.
+
+1. **Requirements.** This guide is intended and tested on Ubuntu 19.04.
+    However, many other Linux distributions will also do fine.
+    For desec-stack, [docker](https://docs.docker.com/install/linux/docker-ce/ubuntu/) and [docker-compose](https://docs.docker.com/compose/install/) are required.
+    Further tools that are required to start hacking are git and curl.
+    Recommended, but not strictly required for desec-stack development is to use certbot along with Let's Encrypt and PyCharm.
+    jq, httpie, python3-venv, libmysqlclient-dev and python3-dev are useful if you want to follow this guide.
+    To install everything you need for this guide except docker and docker-compose, use
+
+        sudo apt install git curl httpie jq python3-venv libmysqlclient-dev python3-dev
+
+1. **Get the code.** Clone this repository to your favorite location.
+
+        git clone git@github.com:desec-io/desec-stack.git
+
+1. **Obtain Domain Names.** To run desec-stack, this guide uses a subdomain of dedyn.io provided by desec.io.
+    Install the httpie software, `sudo apt install httpie` to run the following commands.
+    1. Register a deSEC user account.
+
+            http POST https://desec.io/api/v1/auth/users/ email:='"you@example.com"' password:='"secret"'
+            http POST https://desec.io/api/v1/auth/token/login/ email:='"you@example.com"' password:='"secret"'
+
+        The deSEC API will reply with an authentication token to the second request, similar to 
+
+            {
+                "auth_token": "i+T3b1h/OI+H9ab8tRS98stGtURe"
+            }
+
+        Setup a shell variable that holds the authentication token for future use:
+
+            TOKEN=i+T3b1h/OI+H9ab8tRS98stGtURe
+
+        Check your email and follow the instructions for completing the registration.
+
+    2. Register a dedyn.io subdomain to run your desec-stack on it and set up the IP addresses.
+        For this guide, we assume `example.dedyn.io`. Register it with:
+
+            http POST https://desec.io/api/v1/domains/ Authorization:"Token $TOKEN" name:='"example.dedyn.io"'
+
+        If you receive an answer that is different from status code 201, chances are that the name you chose is already taken by someone else.
+        In that case, repeat the last step with a new name.
+        To setup the necessary IP address records, we create a couple of A and AAAA records that point to localhost.
+        As preparation, create a JSON file `dns.json` with the following content defining the DNS setup for desec-stack:
+
+            [
```

Comment:
get.desec subname is missing. I think it would be easier and more future-proof to just set a CNAME on `*.desec` pointing `desec.$DOMAIN.`.

Reply:
CNAME is not straightforward as this is the content of a file and we'd somehow have to deal with substituting `$DOMAIN` etc. So, let's keep the A/AAAA records. I still cleaned it up a bit, by using wildcards.